build-tools/mono-runtimes/mono-runtimes.targets

```diff
@@ -200,7 +200,7 @@
 
   <Target Name="_Autogen"
       DependsOnTargets="_SetAutogenShTimeToLastCommitTimestamp"
-      Inputs="$(MonoSourceFullPath)\autogen.sh"
+      Inputs="$(MonoSourceFullPath)\autogen.sh;$(MonoSourceFullPath)\configure.ac"
```

Comment:
Why add `configure.ac`?

The reason for `autogen.sh` being listed as an input is *not* because we at some point execute `autogen.sh`. The reason is because `//Target/@Inputs` and `//Target/@Outputs` are based entirely on *timestamps* (like make targets, really), and the dependent target `_SetAutogenShTimeToLastCommitTimestamp` sets the timestamp of `autogen.sh` to the time of the HEAD commit within `external/mono`.

The logic being that when `external/mono` is updated (e.g. via `make prepare`), `_SetAutogenShTimeToLastCommitTimestamp` will set the timestamp of `autogen.sh` to the timestamp of the "new" HEAD commit, which will presumably be newer than the `configure` output file, which will cause `_Autogen` to be re-run.

Repeating myself, this is all *timestamp* driven.

There is no need to add `configure.ac` here. If there *is* a need to add `configure.ac` here, then that strongly implies you want *something else*, e.g. "rebuild whenever *any* file in mono is touched." (Otherwise why limit to just `configure.ac`?)

Reply:
If you modify `configure.ac`, you want to rerun this step, and depending only on `autogen.sh` is not enough since most of the logic is in `configure.ac` and not in `autogen.sh`. It's also the way we do it in `sdks/builds/`.

Reply:
@luhenry: that argument doesn't "scale". If you touch `external/mono/mono/mini/aot-compiler.c`, shouldn't everything be rebuilt? What about `external/mono/mcs/class/corlib/System/Array.cs`?

There are only three reasonable answers:

1. Do what we're doing. If you want to trigger a mono rebuild, you either:
     1.  [*work around the system*](https://github.com/xamarin/xamarin-android/blob/f573af9cb25fa7bb74b960a946ee6b7f10ddcee5/Documentation/DevelopmentTips.md#the-long-way)

                make -C build-tools/mono-runtimes/obj/Debug/armeabi-v7a

            or

     2. Add a new commit to `external/mono`, thus causing `_SetAutogenShTimeToLastCommitTimestamp` to kick in

2. Add a dependency on *everything*

    ```xml
    <ItemGroup>
      <_MonoSource Include="$(MonoSourceFullPath)\**\*.c" />
      <_MonoSource Include="$(MonoSourceFullPath)\**\*.cs" />
      <_MonoSource Include="$(MonoSourceFullPath)\autogen.sh" />
      <_MonoSource Include="$(MonoSourceFullPath)\configure.ac" />
      <!-- ...? -->
    <ItemGroup>
    <Target Name="_Autogen"
        Inputs="@(_MonoSource)"
      ...
    ```

    This has the advantage of being *really elegant*: touch ~anything in mono, and we rebuild! What more could you want?

    The problem here is *cost*: depending on filesystem/OS/etc., asking MSBuild to evaluate `@(_MonoSource)` will add 4+ *minutes* to your build time, for **EVERY** build. **AT EVERY LEVEL**. (Because most projects depend directly or indirectly on `mono-runtimes.mdproj`.)

    I *personally* consider this solution *unacceptable*: `mono` isn't updated often enough in normal development to make the *added cost* of 4 *minutes* per build beneficial in any way.

3. Punt on the entire thing. This is @grendello's "let's just use `premake` to build mono/etc!" idea.

(3) will be investigated, but likely won't exist before we want to merge this PR.

I have not yet dreamed up a 4th option, but I'm open to suggestions.

Reply:
I am not saying we should "scale" that approach to all files in Mono, just to `configure` and `configure.ac`.